corehq/apps/users/util.py

```diff
@@ -507,3 +507,61 @@ def is_username_available(username):
 
     exists = user_exists(username)
     return not exists.exists
+
+
+class SimpleProgressHelper(object):
+
+    def __init__(self, identifier):
+        self.identifier = identifier
+
+    @property
+    def key_expiry(self):
+        return 60 * 60 * 24
+
+    @property
+    def total_key(self):
+        return f'progress-total:{self.identifier}'
+
+    @property
+    def current_key(self):
+        return f'progress-current:{self.identifier}'
+
+    def set_initial(self, total, current=0):
+        cache.set(self.total_key, total)
+        cache.set(
+            self.current_key,
+            current,
+        )
+        self._set_expiry(self.total_key)
+        self._set_expiry(self.current_key)
+
+    def increment(self, increment_value=1):
+        self.set_current(self.current + increment_value)
+
+    def set_current(self, val):
+        cache.set(self.current_key, val)
+
+    @property
+    def total(self):
+        return cache.get(self.total_key)
+
+    @property
+    def current(self):
+        return cache.get(self.current_key)
+
+    @property
+    def percentage_complete(self):
```

Comment:
Hmm, the whole body of the method is inside a conditional, which is not preferred. I'd change the first conditional to
```python
if not (self.total and self.current):
   return
...
```

Reply:
Hmm, on second thought. The responsibility of `percentage_complete` is to return some progress value. Returning `None` seems surprising and leave the caller with the responsibility to handle/interpret this.

I see why you did this when looking at the `clearing_process_busy` method on the calling side, but a cleaner solution would be to expose a `clearing_process_busy` method on the ProgressManager rather so that the caller is not responsible to do this. This also cleans up the `percentage_complete` so that no `None` values will surprise the caller

Reply:
Mmm...yeah, good point.